nav2_graceful_motion_controller/src/path_handler.cpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2023 Alberto J. Tudela Roldán
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <string>
+#include <limits>
+#include <memory>
+#include <vector>
+#include <utility>
+
+#include "nav2_core/controller_exceptions.hpp"
+#include "nav2_util/node_utils.hpp"
+#include "nav2_util/geometry_utils.hpp"
+#include "nav_2d_utils/tf_help.hpp"
+#include "nav2_graceful_motion_controller/path_handler.hpp"
+
+namespace nav2_graceful_motion_controller
+{
+
+using nav2_util::geometry_utils::euclidean_distance;
+
+PathHandler::PathHandler(
+  tf2::Duration transform_tolerance,
+  std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros)
+: transform_tolerance_(transform_tolerance), tf_buffer_(tf), costmap_ros_(costmap_ros)
+{
+}
+
+nav_msgs::msg::Path PathHandler::transformGlobalPlan(

```

Comment:
I assume most of this file is lifted from another controller - which one and did you make any modifications? Makes reviewing this easier if I know what is different or if its all the same as another

Reply:
Yes, it's from the regulated pure pursuit controller without any modifications, if there are any.